frontend/public/components/secrets/create-secret/types.ts

```diff
@@ -1,25 +1,28 @@
-import { SecretType } from '.';
+import { SecretType } from './const';
 
 export type SecretStringData = { [key: string]: string };
 
 type SecretChangeData = {
-  stringData: SecretStringData;
+  stringData?: SecretStringData;
   base64StringData?: SecretStringData;
 };
 
-export type KeyValueEntryFormState = {
+export type OpaqueDataEntry = {
   isBase64?: boolean;
   isBinary?: boolean;
```

Comment:
I don't think we need both of these. 
```suggestion
  isBinary?: boolean;
```